tool/tctl/common/collection.go

```diff
@@ -1010,8 +1010,8 @@ type samlIDPServiceProviderCollection struct {
 
 func (c *samlIDPServiceProviderCollection) resources() []types.Resource {
 	r := make([]types.Resource, len(c.serviceProviders))
-	for _, resource := range c.serviceProviders {
-		r = append(r, resource)
+	for i, resource := range c.serviceProviders {
+		r[i] = resource
 	}
```

Comment:
Nit:
```suggestion
	copy(r, c.serviceProviders)
```

Reply:
So unfortunately the copy doesn't work here because the `types.Resource` array doesn't match here. I'll have to revert this change.